script/install.sh

```diff
@@ -619,13 +627,13 @@ uninstall_compiler() {
         fatal "$1 is not installed in $path"
     fi
     log "Removing $path/$1"
-    rm -rf "$path/$1"
+    rm -rf "${path:-.}/$1"
 }
 
 list_compilers() {
     check_tools egrep
     if [ -d "$path" ]; then
-        ls "$path" | egrep -v '^dub|^install\.sh|^d-keyring\.gpg'
+        find "$path" -maxdepth 1 -printf "%f\n" 1 | grep -Ev '^dub|^install\.sh|^d-keyring\.gpg'
```

Comment:
As I understand, `ls`'s output is not meant to be parseable, it's meant for humans. The particular warning here is [SC2012](https://github.com/koalaman/shellcheck/wiki/SC2012), but there are some other related warnings related to piping `ls`.

Note that this fix was incorrect and suboptimal; in the latest version it is all one `find` invocation, which also allows us to drop `grep`/`egrep`.